ui/src/app/users/table.tsx

```diff
@@ -1,43 +1,100 @@
+import { useState, useContext } from "react"
 import { AsideContext } from "../aside"
 import { UserEntry } from "../types"
-import Row from "./row"
+import { useMutation, useQueryClient } from "react-query"
+import { Button, ContextualMenu, MainTable, Panel } from "@canonical/react-components";
+import { ConfirmationModalData, UsersConfirmationModal, ChangePasswordModalData, ChangePasswordModal } from "./components"
+import { useAuth } from "../auth/authContext"
+import { deleteUser } from "../queries"
 
 type TableProps = {
     users: UserEntry[]
 }
 
-export function UsersTable({ users: rows }: TableProps) {
+export function UsersTable({ users }: TableProps) {
+    const auth = useAuth()
     const { isOpen: isAsideOpen, setIsOpen: setAsideIsOpen } = useContext(AsideContext)
     const asideContext = useContext(AsideContext)
-    const [actionsMenuExpanded, setActionsMenuExpanded] = useState<number>(0)
+    const [confirmationModalData, setConfirmationModalData] = useState<ConfirmationModalData | null>(null)
+    const [changePasswordModalData, setChangePasswordModalData] = useState<ChangePasswordModalData>(null)
+    const queryClient = useQueryClient()
+    const deleteMutation = useMutation(deleteUser, {
+        onSuccess: () => queryClient.invalidateQueries('users')
+    })
+    const handleDelete = (id: string, username: string) => {
+        setConfirmationModalData({
+            warningText: `Deleting user: "${username}". This action cannot be undone.`,
+            onMouseDownFunc: () => {
+                const authToken = auth.user ? auth.user.authToken : "";
+                deleteMutation.mutate({ id: id, authToken });
+            }
+        });
+    };
+    const handleChangePassword = (id: string, username: string) => {
+        setChangePasswordModalData({ "id": id, "username": username })
+    }
+
     return (
-        <div className="p-panel">
-            <div className="p-panel__header is-sticky">
-                <h4 className="p-panel__title">Users</h4>
-                <div className="p-panel__controls">
-                    {rows.length > 0 && <button className="u-no-margin--bottom p-button--positive" aria-label="add-csr-button" onClick={() => { asideContext.setExtraData(null); setAsideIsOpen(true) }}>Create New User</button>}
-                </div>
-            </div>
-            <div className="p-panel__content">
-                <div className="u-fixed-width">
-                    <table id="csr-table" aria-label="Certificate Requests Table" className="p-table--expanding">
-                        <thead>
-                            <tr>
-                                <th>ID</th>
-                                <th>Username</th>
-                                <th className="has-overflow">Actions</th>
-                            </tr>
-                        </thead>
-                        <tbody>
+        <Panel
+            stickyHeader
```

Comment:
is the stickyHeader value supposed to be a boolean?

Reply:
Yes sir

```
type HeaderProps<L = LogoDefaultElement> = ExclusiveProps<{
    /**
     * This prop can be used to replace the header area of the panel when the default implementation is not sufficient.
     */
    header: ReactNode;
}, {
    /**
     * Content that will be displayed in the controls area.
     */
    controls?: ReactNode;
    /**
     * Classes that will be applied to the controls element.
     */
    controlsClassName?: string;
    /**
     * Whether the header should be sticky.
     */
    stickyHeader?: boolean;
    /**
     * The panel toggle attributes.
     */
    toggle?: PanelToggle;
}
```